File: src/main/java/dev/slne/protect/bukkitold/region/visual/visualizer/color/VisualizerColor.java
package dev.slne.protect.bukkitold.region.visual.visualizer.color;

import com.github.retrooper.packetevents.protocol.world.states.WrappedBlockState;
import com.github.retrooper.packetevents.protocol.world.states.type.StateTypes;
import dev.slne.surf.surfapi.core.api.util.BlockStateFactory;

public enum VisualizerColor {

  OWNING(BlockStateFactory.of(StateTypes.LIME_STAINED_GLASS)),
  MEMBER(BlockStateFactory.of(StateTypes.LIGHT_BLUE_STAINED_GLASS)),
  NOT_OWNING(BlockStateFactory.of(StateTypes.RED_STAINED_GLASS));

  private final WrappedBlockState blockState;

  /**
   * Create a new color
   *
   * @param blockState the block state
   */
  VisualizerColor(WrappedBlockState blockState) {
    this.blockState = blockState;
  }

  /**
   * Returns the id of the color
   *
   * @return the id
   */
  public WrappedBlockState getBlockState() {
    return blockState;
  }

}

File: src/main/kotlin/dev/slne/surf/protect/paper/region/visual/visualizer/color/VisualizerColor.kt
package dev.slne.surf.protect.paper.region.visual.visualizer.color

import com.sk89q.worldguard.protection.regions.ProtectedRegion
import org.bukkit.block.BlockType
import org.bukkit.block.data.BlockData
import org.bukkit.entity.Player

enum class VisualizerColor(val blockData: BlockData) {
    OWNING(BlockType.LIME_STAINED_GLASS.createBlockData()),
    MEMBER(BlockType.LIGHT_BLUE_STAINED_GLASS.createBlockData()),
    NOT_OWNING(BlockType.RED_STAINED_GLASS.createBlockData());

    companion object {
        fun selectColor(player: Player, region: ProtectedRegion): VisualizerColor = when {
            region.owners.contains(player.uniqueId) -> OWNING
            region.members.contains(player.uniqueId) -> MEMBER
            else -> NOT_OWNING
        }
    }
}
